x-pack/plugins/security_solution/common/detection_engine/transform_actions.ts

```diff
@@ -12,25 +12,29 @@ import type { RuleAlertAction } from './types';
 export const transformRuleToAlertAction = ({
   group,
   id,
-  action_type_id, // eslint-disable-line @typescript-eslint/naming-convention
+  action_type_id: actionTypeId,
   params,
+  uuid,
```

Comment:
I don't know how come this has been working :) Probably even though actionTypeId is passed as param, it has always been `undefined`. `NormalizedAlertAction` is for UI, we strip out some fields before sending actions to UI and add them back before updating. I'll try to update the schema accordingly.

Reply:
We use this function before we pass an action to the framework:

- server-side: before we call `rulesClient.update` or `rulesClient.create`
- client-side: before we pass actions to the Framework's components

I believe instead of using `RuleActionWithOptionalUuidCamel` as the return type we should use some type provided by the framework itself.

Same for `transformAlertToRuleAction`: it should accept some type defined in the framework.

Reply:
This is not that easy unfortunately :)

Update and create methods expect `NormalizedAlertAction` which have optional 'uuid' field but no `actionTypeId` field.

But UI uses `RuleAction` which have required `actionTypeId` and `uuid`

So, one function to transform from `RuleAlertAction` to `NormalizedAlertAction` and/or `RuleAction` is not possible. 

Therefore i added uuid as optional everywhere in security solution (Update, Create  params, Response interface etc.) Just casted actions in  [x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/helpers.tsx](https://github.com/elastic/kibana/pull/148038/files#diff-c718bc61f925a15ede964cff08fcaf559a2cfe86f313be889867876faebddb81) as RuleAction.

I'll leave adding new interface and schemas to use for rulesClient's update/create etc. methods to you. Otherwise this PR would balloon again.

Reply:
After converting `uuid` in RuleAction to an optional field, these functions work as they were.